db/upgrade.php

```diff
@@ -388,5 +388,25 @@ function xmldb_auth_saml2_upgrade($oldversion) {
         upgrade_plugin_savepoint(true, 2022031503, 'auth', 'saml2');
     }
 
+    if ($oldversion < 2023100300) {
+        $protocols = get_config('auth_saml2', 'assertionsconsumerservices');
+        $protocols = explode(',', $protocols);
+        $a = array_search('urn:oasis:names:tc:SAML:1.0:profiles:browser-post', $protocols);
+        $b = array_search('urn:oasis:names:tc:SAML:1.0:profiles:artifact-01', $protocols);
+
+        if ($a) {
```

Comment:
Can this get a quick explainer comment for anyone digging into the upgrade behaviours